### ISSUE TITLE
[Bug]: GIBBON information gain underflows to exactly zero for moderately large max-value samples

### ISSUE THREAD

**Issue by pedromsferraz** [meta-pytorch/botorch#3368]
### What happened?

`qLowerBoundMaxValueEntropy` (and `qMultiFidelityLowerBoundMaxValueEntropy`, which delegates to it) returns exactly `0.0` for most candidates whenever the sampled max values sit well above the posterior mean.

The information gain is computed as `acq = -0.5 * (1 - u).clamp_min(CLAMP_LB).log()` with `u = rho2 * r * (gamma + r)`, `r = phi(gamma) / Phi(gamma)` and `gamma = (m* - mu) / sigma`. Since `u` decays like `exp(-gamma^2 / 2)`, `1 - u` rounds to exactly 1 once `u` drops below half an ulp: past `gamma = 6.0` in float32 and `8.7` in float64 at `rho2 = 1`, earlier for smaller `rho2` (first script below). The true value is far above the smallest representable float (e.g. `1.3e-31` at `gamma = 12`), so this is cancellation in `1 - u`, not the value being unrepresentable.

Because the Gumbel max-value sampler assumes independent candidates, `m*` grows with the candidate-set size `N` while the posterior at a given point does not, so this gets worse as the candidate set grows. On a fitted `SingleTaskGP` with 200 training points in 6 dimensions (second script), 53.1% of a 1000-point test set is exactly zero with a 10k candidate set in float32, rising to 67.0% at 100k, while the median `gamma` goes from 5.85 at `N = 1e3` to 8.04 at `N = 1e5`. Double precision only shifts the cliff (36.6% at 100k).

The result is a flat acquisition surface: `optimize_acqf` has no gradient signal over most of the domain, and any downstream ranking by acquisition value degenerates into an arbitrary tie-break.

Expected: a small but positive information gain, as the formula gives.

Related context: #89 (where the `CLAMP_LB` floors come from), #1814 (NaNs in the `q > 1` term; the thread notes the gain "can be pretty tiny" late in BO), #2926 / #2930 (a different bug in the same method, already fixed).

### Please provide a minimal, reproducible example of the unexpected behavior.

Formula level: feeds a grid of `gamma` straight into `_compute_information_gain` (unit variances, so `rho2` is exactly as given) and compares with a 60-digit mpmath reference (`pip install mpmath`).

```python
"""GIBBON's information-gain expression vs a high-precision reference.

Feeds a grid of gamma = (m* - mean) / std straight into BoTorch's own
`_compute_information_gain` (unit variances, so rho2 is exactly as given) and
compares with mpmath. Requires `pip install mpmath`.
"""
import math
from types import SimpleNamespace

import mpmath
import numpy as np
import torch
from botorch.acquisition.max_value_entropy_search import qLowerBoundMaxValueEntropy


def botorch_ig(gamma, rho2, dtype):
    n = len(gamma)
    variance_m = torch.ones(n, 1, 1, dtype=dtype)
    stand_in = SimpleNamespace(
        model=SimpleNamespace(
            posterior=lambda X, observation_noise=True, posterior_transform=None: SimpleNamespace(
                variance=variance_m
            )
        ),
        posterior_max_values=torch.zeros(1, 1, dtype=dtype),
        posterior_transform=None,
        X_pending=None,
    )
    out = qLowerBoundMaxValueEntropy._compute_information_gain(
        stand_in,
        X=torch.zeros(n, 1, 1, dtype=dtype),
        mean_M=torch.tensor(-gamma, dtype=dtype).reshape(n, 1),   # m* = 0, so gamma = -mean_M
        variance_M=torch.ones(n, 1, dtype=dtype),
        covar_mM=torch.full((n, 1, 1), math.sqrt(rho2), dtype=dtype),
    )
    return out.reshape(-1).double().numpy()


def exact_ig(gamma, rho2):
    mpmath.mp.dps = 60
    out = []
    for g in gamma:
        g = mpmath.mpf(float(g))
        r = mpmath.npdf(g) / mpmath.ncdf(g)
        out.append(float(-mpmath.log1p(-mpmath.mpf(rho2) * r * (g + r)) / 2))
    return np.array(out)


gamma = np.round(np.arange(0.0, 16.0 + 1e-9, 0.1), 6)
print(f"{'rho2':>7} {'dtype':>8} | last gamma with nonzero acq | exact value there | exact value at gamma=12")
for rho2 in (1.0, 0.5, 1e-1, 1e-2, 1e-3):
    exact = exact_ig(gamma, rho2)
    for dtype in (torch.float32, torch.float64):
        vals = botorch_ig(gamma, rho2, dtype)
        last = gamma[vals > 0].max()
        print(f"{rho2:>7g} {str(dtype).replace('torch.',''):>8} | {last:>27.1f} | "
              f"{exact[gamma == last][0]:>17.2e} | {exact[gamma == 12.0][0]:.2e}")
```

End to end, ordinary GP fit:

```python
"""GIBBON returns exactly 0 for most candidates on a routine GP fit."""
import torch
from botorch.acquisition.max_value_entropy_search import qLowerBoundMaxValueEntropy
from botorch.fit import fit_gpytorch_mll
from botorch.models import SingleTaskGP
from gpytorch.mlls import ExactMarginalLogLikelihood


def run(dtype, n_train=200, n_candidates=10_000, d=6, seed=0):
    torch.manual_seed(seed)
    tkwargs = {"dtype": dtype}
    train_X = torch.rand(n_train, d, **tkwargs)
    train_Y = torch.sin(8 * train_X).sum(-1, keepdim=True) + 0.01 * torch.randn(n_train, 1, **tkwargs)
    model = SingleTaskGP(train_X, train_Y)
    fit_gpytorch_mll(ExactMarginalLogLikelihood(model.likelihood, model))
    candidate_set = torch.rand(n_candidates, d, **tkwargs)
    torch.manual_seed(seed)
    acqf = qLowerBoundMaxValueEntropy(model, candidate_set)
    X = torch.rand(1000, 1, d, **tkwargs)
    with torch.no_grad():
        vals = acqf(X)
        post = model.posterior(X.squeeze(1))
        mu, sigma = post.mean.reshape(-1), post.variance.sqrt().reshape(-1)
    mstar = acqf.posterior_max_values.reshape(-1)
    gamma = (mstar.reshape(1, -1) - mu.reshape(-1, 1)) / sigma.reshape(-1, 1)
    print(f"{str(dtype):>13}  N={n_candidates:>7,}  zero fraction={float((vals == 0).float().mean()):.3f}"
          f"  median gamma={float(gamma.median()):5.2f}  max acq={float(vals.max()):.3e}")


for dtype in (torch.float32, torch.float64):
    for n_candidates in (1_000, 10_000, 100_000):
        run(dtype, n_candidates=n_candidates)
```

### Please paste any relevant traceback/logs produced by the example provided.

```shell
First script:


   rho2    dtype | last gamma with nonzero acq | exact value there | exact value at gamma=12
      1  float32 |                         6.0 |          1.82e-08 | 1.29e-31
      1  float64 |                         8.7 |          6.36e-17 | 1.29e-31
    0.5  float32 |                         5.9 |          1.62e-08 | 6.44e-32
    0.5  float64 |                         8.7 |          3.18e-17 | 6.44e-32
    0.1  float32 |                         5.6 |          1.73e-08 | 1.29e-32
    0.1  float64 |                         8.5 |          3.47e-17 | 1.29e-32
   0.01  float32 |                         5.1 |          2.29e-08 | 1.29e-33
   0.01  float64 |                         8.2 |          4.10e-17 | 1.29e-33
  0.001  float32 |                         4.7 |          1.50e-08 | 1.29e-34
  0.001  float64 |                         7.9 |          4.42e-17 | 1.29e-34


Second script:


torch.float32  N=  1,000  zero fraction=0.407  median gamma= 5.85  max acq=1.693e-01
torch.float32  N= 10,000  zero fraction=0.531  median gamma= 6.98  max acq=3.521e-02
torch.float32  N=100,000  zero fraction=0.670  median gamma= 8.04  max acq=4.472e-03
torch.float64  N=  1,000  zero fraction=0.162  median gamma= 5.85  max acq=1.282e-01
torch.float64  N= 10,000  zero fraction=0.251  median gamma= 7.05  max acq=2.413e-02
torch.float64  N=100,000  zero fraction=0.366  median gamma= 8.03  max acq=3.119e-03
```

### BoTorch Version

0.18.2.dev29+g1913b3897 (current `main`)

### Python Version

3.14.3 (torch 2.14.0, gpytorch 1.15.2)

### Operating System

macOS 15.8 (arm64, CPU)

### (Optional) Describe any potential fixes you've considered to the issue outlined above.

Assemble `u` in log space and finish with `log1p`, so `1 - u` is never formed:

```python
log_ratio = standard_normal_log_hazard(-normalized_mvs)  # log(phi(gamma) / Phi(gamma))
finfo = torch.finfo(X.dtype)
log_gamma_plus_ratio = (normalized_mvs + log_ratio.exp()).clamp_min(finfo.tiny).log()
u = rhos_squared * (log_ratio + log_gamma_plus_ratio).exp()
acq = -0.5 * torch.log1p(-u.clamp_max(1 - finfo.eps / 2))
```

`standard_normal_log_hazard` is already in `botorch/utils/probability/utils.py`. Checked against mpmath on `gamma` in `[-8, 20]` and `rho2` in `{1, 0.1, 1e-3}`: nonzero up to `gamma` 13.9-14.4 in float32 and past 20 in float64, max relative error 1.7e-05 (float32) and 5.7e-14 (float64) for values in each dtype's normal range. It also removes two existing inaccuracies: the `clamp_min(1e-8)` on `Phi(gamma)`, which makes the function return a *negative* information gain for `gamma < -5.6` (e.g. -1.08 at `gamma = -5.7`, `rho2 = 1`, float64, where the true value is 1.82), and the cap at about 9.2 nats from the clamp on `inner_term`.

This mirrors what LogEI did for improvement-based acquisitions: Ament et al., "Unexpected Improvements to Expected Improvement" (NeurIPS 2023), note that their approach could help with information-based acquisition functions prone to similar numerical problems, and this is one such case.

Two things I deliberately left out, happy to follow up if you want them: (1) for float32 models the gain can be genuinely below float32's range when `gamma > 14`, which only a float64 evaluation or a log-valued acquisition would address; (2) a log-valued GIBBON variant in the style of `qLogNEI`.

### Pull Request

Yes

### Code of Conduct

- [x] I agree to follow BoTorch's Code of Conduct
